gym/lib/gym/code_signing_mapping.rb

```
@@ -0,0 +1,136 @@
+          # It's a relative path from the workspace file
+          File.join(File.expand_path("..", self.project.path), current_match.first)
+        end.find_all do |current_match|
+          # We're not interested in a `Pods` project, as it doesn't contain any relevant
+          # information about code signing
+          !current_match.end_with?("Pods/Pods.xcodeproj")
+        end
+
+        return @_project_paths
+      else
+        # Return the path as an array
+        return @_project_paths = [self.project.path]
+      end
+    end
+
+    def detect_project_profile_mapping
+      provisioning_profile_mapping = {}
+
+      self.project_paths.each do |project_path|
+        UI.verbose("Parsing project file '#{project_path}' to find selected provisioning profiles")
+
+        begin
+          project = Xcodeproj::Project.open(project_path)
+          project.targets.each do |target|
+            target.build_configuration_list.build_configurations.each do |build_configuration|
+              current = build_configuration.build_settings
+
+              bundle_identifier = current["PRODUCT_BUNDLE_IDENTIFIER"]
+              provisioning_profile_specifier = current["PROVISIONING_PROFILE_SPECIFIER"]
+              provisioning_profile_uuid = current["PROVISIONING_PROFILE"]
+              if provisioning_profile_specifier.to_s.length > 0
+                provisioning_profile_mapping[bundle_identifier] = provisioning_profile_specifier
+              elsif provisioning_profile_uuid.to_s.length > 0
+                provisioning_profile_mapping[bundle_identifier] = provisioning_profile_uuid
+              end
+            end
+          end
+        rescue => ex
+          # We catch errors here, as we might run into an exception on one included project
+          # But maybe the next project actually contains the information we need
+          if Helper.xcode_at_least?("9.0")
```

If Xcode is < 9, should we re-raise the error here?